### ISSUE TITLE
[DetailsList] ariaLabelForSelectAllCheckbox adds an aria-describedby instead of a aria-label - which results to the select all checkbox to not have a name property

### ISSUE THREAD

**Issue by satbai** [OfficeDev/office-ui-fabric-react#4980]
### Bug Report

- __Package version(s)__: latest
- __Browser and OS versions__: all

#### Priorities and help requested (not applicable if asking question):
Are you willing to submit a PR to fix? No
Requested priority: Normal

#### Describe the issue:
Adding ariaLabelForSelectAllCheckbox on the DetailsList component will actually add an aria-describedby instead of aria-label, therefore the name property of the select all checkbox will not get added and the element does not meet the Microsoft accessibility standard.

#### Actual behavior:
![image of DOM showing aria-label and aria-describedby values](https://user-images.githubusercontent.com/24945574/40514330-947004aa-5f5d-11e8-826a-871d51775b96.png)

#### Expected behavior:
ariaLabelForSelectAllCheckbox should add an aria label that will provide a name property for the element. 

Alternatively, the DetailsList component could add a default aria-label for the element if it's not provided through the props. "Select all" would be a good option and cover most of the cases.

### If applicable, please provide a codepen repro:
https://developer.microsoft.com/en-us/fabric#/components/detailslist

**Comment by ThomasMichon** [comment]
@satbai We can't use an `aria-label` here because the `aria-label` will then also be read as the name of the column, which is not desirable. The `aria-describedby` tells the user what the checkbox will do, while the checkbox's own `aria-label` (settable via `ariaLabelForSelectionColumn`) becomes the column header name.

**Comment by natalieethell** [comment]
Thanks for your issue! As Thomas mentioned above, if you'd like to specifically set the `aria-label`, you can utilize `ariaLabelForSelectionColumn`. We don't have a default for `ariaLabelForSelectionColumn` due to localization concerns, but we do have it set on our [DetailsList demo](https://developer.microsoft.com/en-us/fabric#/components/detailslist/variablerowheights) pages.

![image](https://user-images.githubusercontent.com/14134000/49904342-b1303280-fe1e-11e8-8b46-c52901a77cc3.png)

Feel free to let us know if you have any other considerations.
